nototools/sample_with_font.py

```diff
@@ -0,0 +1,184 @@
+
+Often in bug reports people will describe text using character names.
+Replicating the sample text they describe can be a bit tedious.  This
+lets you interactively search characters in the font by name to assemble
+a string and save it to a file."""
+
+import argparse
+import codecs
+import readline
+
+from nototools import coverage
+from nototools import unicode_data
+
+from fontTools import ttLib
+
+def _help():
+  print ('enter a string to match or one of the following:\n'
+         '  \'quit\' to exit,\n'
+         '  \'help\' to show these options,\n'
+         '  \'names\' for names,\n'
+         '  \'dump\' to dump the current text,\n'
+         '  \'clear\' to clear the current text,\n'
+         '  \'write\' to be prompted for a filename to write the text to.')
+
+
+def _build_text(name_map, initial_text=''):
+  text = initial_text
+  print 'build text using map of length %d' % len(name_map)
+  while True:
+    line = raw_input('> ')
+    if not line:
+      continue
+    if line == 'quit':
+      break
+    if line == 'help':
+      _help()
+      continue
+    if line == 'names':
+      print 'names:\n  ' + '\n  '.join(sorted(name_map.keys()))
+      continue
+    if line == 'dump':
+      print 'dump: \'%s\'' % text
+      for cp in text:
+        print '%06x %s' % (ord(cp), unicode_data.name(cp))
+      continue
+    if line == 'clear':
+      text = ''
+      continue
+    if line == 'write':
+      line = raw_input('file name> ')
+      if line:
+        _write_text(line, text)
+      continue
+
+    matches = []
+    for name, cp in sorted(name_map.iteritems()):
+      if line in name:
+        matches.append(name)
+    if not matches:
+      print 'no match for "%s"'% line
+      continue
+
+    if len(matches) == 1:
+      print matches[0]
+      text += unichr(name_map[matches[0]])
+      continue
+
+    # if we match a full line, then use that
+    if line in matches:
+      print line
+      text += unichr(name_map[line])
+      continue
+
+    new_matches = []
+    for m in matches:
+      if line in m.split(' '):
+        new_matches.append(m)
+
+    # if we match a full word, and only one line has this full word, use that
+    if len(new_matches) == 1:
+      print new_matches[0]
+      text += unichr(name_map[new_matches[0]])
+      continue
+
+    select_multiple = True
+    while select_multiple:
+      print 'multiple matches:\n  ' + '\n  '.join(
+          '[%2d] %s' % (i, n) for i, n in enumerate(matches))
+      while True:
+        line = raw_input('0-%d or q to skip> ' % (len(matches) - 1))
+        if line == 'q':
+          select_multiple = False
+          break
+        try:
+          n = int(line)
+          break
+        except ValueError:
+          continue
+
+      if not select_multiple: # q
+        break
+
+      if n < 0 or n >= len(matches):
+        print '%d out of range' % n
+        continue
+
+      text += unichr(name_map[matches[n]])
+      select_multiple = false
```

Comment:
Man, I'm just batting 1000 on this aren't I.

On Thu, Oct 22, 2015 at 10:40 AM, James Godfrey-Kittle <
notifications@github.com> wrote:

> In nototools/sample_with_font.py
> <https://github.com/googlei18n/nototools/pull/105#discussion_r42780054>:
>
> > +          break
> > +        try:
> > +          n = int(line)
> > +          break
> > +        except ValueError:
> > +          continue
> > +
> > +      if not select_multiple: # q
> > +        break
> > +
> > +      if n < 0 or n >= len(matches):
> > +        print '%d out of range' % n
> > +        continue
> > +
> > +      text += unichr(name_map[matches[n]])
> > +      select_multiple = false
>
> False should be capitalized
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlei18n/nototools/pull/105/files#r42780054>.
>